pyvista/core/utilities/fileio.py

```diff
@@ -163,6 +174,24 @@ def set_vtkwriter_mode(vtk_writer: _VTKWriterType, use_binary: bool = True) -> _
     elif isinstance(vtk_writer, _vtk.vtkXMLWriter):
         if use_binary:
             vtk_writer.SetDataModeToBinary()
+            supported = get_args(_CompressionOptions)
+            if compression not in supported:
+                supported_str = ', '.join(
+                    f"'{x}'" if x is not None else '`None`' for x in supported
+                )
+                msg = (
+                    f"Unsupported compression format '{compression}'. "
+                    f'Valid options are {supported_str}.'
+                )
+                raise ValueError(msg)
```

Comment:
Can use our validation methods for this. 
```suggestion
            from pyvista.core import _validation
            _validation.check_contains(supported, must_contain=compression, name='compression')
```
The existing `raise ValueError` test will either need to be updated, or can be removed.